dali/operators/bbox/bbox_rotate.cc

```diff
@@ -0,0 +1,388 @@
+  } else {
+    ExpandToAllCorners<false>(x_coords, y_coords, in_boxes);
+  }
+
+  // Center the coordinates on zero in image coordinates
+  if (bbox_norm) {
+    std::transform(x_coords.begin(), x_coords.end(), x_coords.begin(),
+                   [w = image_wh.x](float elem) { return (elem - 0.5) * w; });
+    std::transform(y_coords.begin(), y_coords.end(), y_coords.begin(),
+                   [h = image_wh.y](float elem) { return (elem - 0.5) * h; });
+  } else {
+    std::transform(x_coords.begin(), x_coords.end(), x_coords.begin(),
+                   [half_w = image_wh.x / 2](float elem) { return elem - half_w; });
+    std::transform(y_coords.begin(), y_coords.end(), y_coords.begin(),
+                   [half_h = image_wh.y / 2](float elem) { return elem - half_h; });
+  }
+
+  // Need to log old wh for expansion correction if needed
+  std::optional<std::vector<vec2>> old_wh;
+  if (mode != Mode::Expand) {
+    old_wh->resize(num_boxes);
+    for (int i = 0; i < num_boxes; ++i) {  // x,y coords are in order x1x2x1x2 y1y1y2y2
+      (*old_wh)[i] = {x_coords[4 * i + 1] - x_coords[4 * i], y_coords[4 * i + 2] - y_coords[4 * i]};
+    }
+  }
+
+  // Apply rotation
+  RotateCorners(x_coords, y_coords, angle);
+
+  // Move coordinates back to [0,{w|h}], updating to new coordinate system
+  if (!keep_size) {
+    const auto old_w = image_wh.x;
+    image_wh.x = std::abs(image_wh.x * std::cos(angle)) + std::abs(image_wh.y * std::sin(angle));
+    image_wh.y = std::abs(image_wh.y * std::cos(angle)) + std::abs(old_w * std::sin(angle));
+  }
+  std::transform(x_coords.begin(), x_coords.end(), x_coords.begin(),
+                 [half_w = image_wh.x / 2](float elem) { return elem + half_w; });
+  std::transform(y_coords.begin(), y_coords.end(), y_coords.begin(),
+                 [half_h = image_wh.y / 2](float elem) { return elem + half_h; });
+
+  // Convert back to bounding boxes
+  auto out_boxes =
+      dali::span(reinterpret_cast<vec4*>(out_box_tensor.raw_mutable_data()), num_boxes);
+  CornersToBoxes(out_boxes, x_coords, y_coords);
+
+  // Apply correction to expansion factor if required
+  if (mode != Mode::Expand) {
+    ApplyExpansionCorrectionToBoxSize(out_boxes, *old_wh, mode == Mode::Halfway);
+  }
+
+  // Clip to image coordinates and remove boxes (and labels) with too large a reduction.
+  const auto num_out_boxes =
+      ClipAndRemoveBoxes(out_boxes, remove_threshold, image_wh, in_labels, out_labels);
+
+  if (!ltrb || bbox_norm) {  // Convert back to xywh and/or normalized format if needed
+    out_boxes = dali::span(out_boxes.data(), num_out_boxes);  // handle if some boxes culled
+    std::for_each(out_boxes.begin(), out_boxes.end(), [image_wh, ltrb, bbox_norm](vec<4>& box) {
+      if (!ltrb) {
+        box.z -= box.x;
+        box.w -= box.y;
+      }
+      if (bbox_norm) {
+        box.x /= image_wh.x;
+        box.y /= image_wh.y;
+        box.z /= image_wh.x;
+        box.w /= image_wh.y;
```

Comment:
I wonder if we really need this level of precision. If the division by the image size is not equal to the (much faster) multiplication by reciprocal, then the original normalized coordinates weren't very precise to start with.